#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Doucment handling of paramters of the test builtin.

--- builtins/test.def~	2004-10-17 17:04:56.000000000 +0200
+++ builtins/test.def	2004-10-17 17:19:38.000000000 +0200
@@ -92,6 +92,9 @@
 Arithmetic binary operators return true if ARG1 is equal, not-equal,
 less-than, less-than-or-equal, greater-than, or greater-than-or-equal
 than ARG2.
+
+See the bash manual page bash(1) for the handling of parameters (i.e.
+missing parameters).
 $END
 
 $BUILTIN [
--- doc/bash.1~	2005-09-09 19:43:43.298296896 +0000
+++ doc/bash.1	2005-09-09 19:46:26.359507824 +0000
@@ -3551,6 +3551,10 @@
 .PP
 Unless otherwise specified, primaries that operate on files follow symbolic
 links and operate on the target of the link, rather than the link itself.
+.PP
+See the description of the \fItest\fP builtin command (section SHELL
+BUILTIN COMMANDS below) for the handling of parameters (i.e.
+missing parameters).
 .sp 1
 .PD 0
 .TP
